src/OMSimulatorLib/System.cpp

```diff
@@ -1590,13 +1620,19 @@ oms_status_enu_t oms::System::delete_(const oms::ComRef& cref)
   {
     auto subsystem = subsystems.find(front);
     if (subsystem != subsystems.end())
+    {
+      deleteAllConectionsTo(cref);
       return subsystem->second->delete_(tail);
+    }
 
     auto component = components.find(front);
     if (component != components.end())
+    {
+      deleteAllConectionsTo(cref);
       return component->second->deleteConnector(tail);
+    }
   }
-
+  logWarning("failed to delete object \"" + std::string(getFullCref()+front) + "\"" + " because the identifier couldn't be resolved to any connector, component, system, or model");
```

Comment:
I guess it should be `std::string(getFullCref() + cref)`. `tail` doesn't necessarily need to be empty.

Reply:
changed the warning message https://github.com/OpenModelica/OMSimulator/pull/758/commits/6bf8b45f0071257e82813e5d62aa24321c2d47e8